.github/workflows/build.yaml

```diff
@@ -93,123 +93,61 @@ jobs:
     needs: workflow-setup
     runs-on: ubuntu-latest
     env:
-      CACHE_KEY_BUILD: ${{ needs.workflow-setup.outputs.CACHE_KEY_BUILD }}
-      GITHUB_REPOSITORY_NAME: ${{ needs.workflow-setup.outputs.GITHUB_REPOSITORY_NAME }}
+      GITHUB_REPOSITORY_OWNER: ${{ needs.workflow-setup.outputs.GITHUB_REPOSITORY_OWNER}}
       UBUNTU_VERSION: ${{ needs.workflow-setup.outputs.UBUNTU_VERSION }}
     steps:
       - name: Git checkout
         uses: actions/checkout@v2
 
-      - name: Try load from cache
-        id: cache-image
-        uses: actions/cache@v2
-        with:
-          path: ${GITHUB_WORKSPACE}/cache
-          key: ${{ env.CACHE_KEY_BUILD}}
-
       - name: Prepare image labels and tags
-        if: steps.cache-image.outputs.cache-hit != 'true'
         id: prep
         shell: bash
         run: |
-          DOCKER_IMAGE=ghcr.io/${{ env.GITHUB_REPOSITORY_NAME }}/node-build
-          TAGS="${DOCKER_IMAGE}:latest,${DOCKER_IMAGE}:${{ env.UBUNTU_VERSION }}"
-          echo ::set-output name=tags::${TAGS}
           echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
 
-      - name: Log into the GitHub Container Registry
-        if: steps.cache-image.outputs.cache-hit != 'true'
-        uses: docker/login-action@v1
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Set up Docker Buildx
-        if: steps.cache-image.outputs.cache-hit != 'true'
-        uses: docker/setup-buildx-action@v1
-
-      - name: Build and push image
-        if: steps.cache-image.outputs.cache-hit != 'true'
-        uses: docker/build-push-action@v2
-        with:
-          context: .
-          file: .github/workflows/build/Dockerfile.${{ env.UBUNTU_VERSION }}
-          no-cache: true
-          push: ${{ github.event_name != 'pull_request' }}
-          tags: ${{ steps.prep.outputs.tags }}
+      - name: Set image labels and tags
+        id: dockermetadata
+        uses: docker/metadata-action@v3
+        with: 
+          images: ghcr.io/${{ env.GITHUB_REPOSITORY_OWNER }}/node-build
+          tags: |
+            type=sha
+            type=raw,${{ env.UBUNTU_VERSION }}
+            type=raw,latest
           labels: |
             org.opencontainers.image.source=${{ github.event.repository.html_url }}
             org.opencontainers.image.created=${{ steps.prep.outputs.created }}
             org.opencontainers.image.revision=${{ github.sha }}
-
-      - name: Touch Cache
-        if: steps.cache-image.outputs.cache-hit != 'true'
-        run: |
-          mkdir -p ${GITHUB_WORKSPACE}/cache
-          touch ${GITHUB_WORKSPACE}/cache/${{ env.CACHE_KEY_BUILD }}
-
-  lint-image:
```

Comment:
If the lint image is not needed anymore the related CI/CD sources and artifacts should be deleted.